### PR TITLE
Rework PlaceholderExpansion page

### LINKED ISSUE #653: Add wiki about setRelationalPlaceholders(Player one, Player two, String text);
### Confirmation

- [X] I checked the Issues and Pull request tab for any existing issues or PRs.
- [X] My Feature Request is for **PlaceholderAPI** and not any expansion or other plugin.

### Type

Minor Code improvement (Won't affect Servers)

### Description

Add wiki about setRelationalPlaceholders(Player one, Player two, String text);
I don't know how to use it, so I can't submit pull request.

### Code Example

_No response_

### Jar file

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by Andre601**
### Type

- [ ] Internal change (Doesn't affect end-user).
- [ ] External change (Does affect end-user).
- [x] Wiki (Changes towards the [Wiki]).
- [ ] Other: __________ 

### Description

Closes #653 

I reworked the entirety of the PlaceholderExpansion page as I wasn't happy with the way it was structured.
Any feedback (Even those for grammar) would be appreciated.

I will merge this once either @Funny-cube or @PiggyPiglet give their OK on this.

[Wiki]: https://github.com/PlaceholderAPI/PlaceholderAPI/wiki

### CODE REVIEW

**Review by Funny-cube — CHANGES_REQUESTED** [review]
Quite a few wording errors,

L61 `Through this method can you set the name` -> `Through this method you can set the name`
L68 `which means it doesn't has to be` -> `which means it doesn't have to be`
L69 `part of a plugin does this not really matter.` -> `part of another plugin does this does not really matter.`
L84 `don't need much more to properly work.` - Suggest rewording, eg `don't require any additional plugins to function.`
L85 Needs rewording, eg. `These expansions have the following additional methods added:`
L133 `its placeholder values will you need to override a few more methods to make everything work as it should.` -> `its placeholder values you will need to override a few more methods to make sure everything  will work as it should.`

Only got up to Line 133, suggest re-reading over everything as there is quite a few typos and words in the wrong order, feel free to request another review after you've done so.

**Comment by Andre601** [comment]
You could've just suggested those directly through GitHub's suggestion thing, but alright.

> L85 Needs rewording, eg. `These expansions have the following additional methods added:`

Not sure about this line. This rewording makes it sound like as if it was a exclusive thing to that type of expansion... Should've added it to the common parts section, which I now do...

**Comment by Funny-cube** [comment]
No idea how the suggestion thing works, don't use GitHub to often.

L85 was just a suggestion, just didn't seem right when I read over it. :)

**Comment by Andre601** [comment]
> No idea how the suggestion thing works, don't use GitHub to often.

"Mark" the Line(s) you want to have changed (click the Blue scare with white plus (click and drag for multi-line selection) and in the opening text field, click that file icon.

It would now print a code block with `suggestion` as highlighting. In that code block, change the text to what you want it to be and finally click "Start a Review".
image

Repeat this for all the text you want to change and in the top right, click "Review Changes", select "Request changes", give an optional comment and submit.

That's the tl;dr of how it works (And you should work more with GitHub imo)

File: wiki/PlaceholderExpansion.md [hunk 1]
```diff
@@ -58,19 +58,27 @@ public class SomeExpansion extends PlaceholderExpansion {
 Let's quickly break down the different methods you had to import.
 
 - #### getAuthor
-  Through this method can you set the name of who created the expansion.
+  Through this method you can set the name of who created the expansion.
```
Comment (Starmism) [inline wiki/PlaceholderExpansion.md:61]:
```suggestion
  This method allows you to set the name of the expansion's author.
```

File: wiki/PlaceholderExpansion.md [hunk 2]
```diff
@@ -58,19 +58,27 @@ public class SomeExpansion extends PlaceholderExpansion {
...
 - #### getIdentifier
   The name that should be used to identify the placeholders for this expansion.  
   The Identifier is the first text after the `%` and before the first `_` (`%identifier_values%`) and can therefore not contain any `_` in it.
   
   If you want to use `_` in your Expansion's name can you override the optional `getName()` method.
```
Comment (Starmism) [inline wiki/PlaceholderExpansion.md:66]:
```suggestion
  If you want to use `_` in your expansion's name, you can override the optional `getName()` method.
```

File: wiki/PlaceholderExpansion.md [hunk 3]
```diff
@@ -58,19 +58,27 @@ public class SomeExpansion extends PlaceholderExpansion {
 Let's quickly break down the different methods you had to import.
 
 - #### getAuthor
-  Through this method can you set the name of who created the expansion.
+  Through this method you can set the name of who created the expansion.
 - #### getIdentifier
   The name that should be used to identify the placeholders for this expansion.  
   The Identifier is the first text after the `%` and before the first `_` (`%identifier_values%`) and can therefore not contain any `_` in it.
```
Comment (Starmism) [inline wiki/PlaceholderExpansion.md:64]:
```suggestion
The identifier is the string after the starting `%` and before the first `_` (`%**identifier**_values%`) and, therefore, cannot contain any `_`s.
```

File: wiki/PlaceholderExpansion.md [hunk 4]
```diff
@@ -58,19 +58,27 @@ public class SomeExpansion extends PlaceholderExpansion {
...
   
   If you want to use `_` in your Expansion's name can you override the optional `getName()` method.
 - #### getVersion
-  This is a String, which means it doesn't has to be a number in itself. The String is used to determine if a new update is available or not, when the expansion is shared on the eCloud.  
-  For Expansions that are part of a plugin does this not really matter.
+  This is a String, which means it doesn't have to be a number in itself. The String is used to determine if a new update is available or not, when the expansion is shared on the eCloud.
```
Comment (Starmism) [inline wiki/PlaceholderExpansion.md:68]:
```suggestion
This is a string, which means it can contain more than just a number. This is used to determine if a new update is available or not when the expansion is shared on the eCloud.
```

File: wiki/PlaceholderExpansion.md [hunk 5]
```diff
@@ -58,19 +58,27 @@ public class SomeExpansion extends PlaceholderExpansion {
...
 - #### getIdentifier
   The name that should be used to identify the placeholders for this expansion.  
   The Identifier is the first text after the `%` and before the first `_` (`%identifier_values%`) and can therefore not contain any `_` in it.
   
   If you want to use `_` in your Expansion's name can you override the optional `getName()` method.
 - #### getVersion
-  This is a String, which means it doesn't has to be a number in itself. The String is used to determine if a new update is available or not, when the expansion is shared on the eCloud.  
-  For Expansions that are part of a plugin does this not really matter.
+  This is a String, which means it doesn't have to be a number in itself. The String is used to determine if a new update is available or not, when the expansion is shared on the eCloud.  
+  For Expansions that are part of another plugin does this does not really matter.
```
Comment (Starmism) [inline wiki/PlaceholderExpansion.md:69]:
```suggestion
For expansions that are part of a plugin, this does not really matter.
```

File: wiki/PlaceholderExpansion.md [hunk 6]
```diff
@@ -58,19 +58,27 @@ public class SomeExpansion extends PlaceholderExpansion {
...
 
 Those are all the neccessary parts for your PlaceholderExpansion.  
 Any other methods that are part of the [`PlaceholderExpansion`][placeholderexpansion] class are optional and will usually not be used, or default to a specific value. Please read the Javadoc comments of those methods for more information.
 
+There are two specific methods which are options, but you should use at least one of them:
```
Comment (Starmism) [inline wiki/PlaceholderExpansion.md:74]:
```suggestion
You must choose between one of these two methods for handling the actual parsing of placeholders:
```

File: wiki/PlaceholderExpansion.md [hunk 7]
```diff
@@ -58,19 +58,27 @@ public class SomeExpansion extends PlaceholderExpansion {
...
+
+- #### onRequest(OfflinePlayer, String)
+  If not set this method will default to calling [`onPlaceholderRequest(Player, String)`](#onplaceholderrequestplayer-string).
```
Comment (Starmism) [inline wiki/PlaceholderExpansion.md:77]:
```suggestion
If not explicitly set, this will automatically call [`onPlaceholderRequest(Player, String)`](#onplaceholderrequestplayer-string).
```

File: wiki/PlaceholderExpansion.md [hunk 8]
```diff
@@ -58,19 +58,27 @@ public class SomeExpansion extends PlaceholderExpansion {
...
+  For Expansions that are part of another plugin does this does not really matter.
 
 Those are all the neccessary parts for your PlaceholderExpansion.  
 Any other methods that are part of the [`PlaceholderExpansion`][placeholderexpansion] class are optional and will usually not be used, or default to a specific value. Please read the Javadoc comments of those methods for more information.
 
+There are two specific methods which are options, but you should use at least one of them:
+
+- #### onRequest(OfflinePlayer, String)
+  If not set this method will default to calling [`onPlaceholderRequest(Player, String)`](#onplaceholderrequestplayer-string).  
+  This method is recommended to use as it allows the usage of `null` and can therefore be used in placeholders that don't need a valid player to be used.
```
Comment (Starmism) [inline wiki/PlaceholderExpansion.md:78]:
```suggestion
This method is recommended as it allows the usage of `null` and can therefore be used in placeholders that don't require a valid player to be used.
```

File: wiki/PlaceholderExpansion.md [hunk 9]
```diff
@@ -58,19 +58,27 @@ public class SomeExpansion extends PlaceholderExpansion {
...
+- #### onPlaceholderRequest(Player, String)
+  If not set this method will return `null` which PlaceholderAPI sees as invalid placeholder.
```
Comment (Starmism) [inline wiki/PlaceholderExpansion.md:80]:
```suggestion
If not set, this method will return `null` which PlaceholderAPI sees as an invalid placeholder.
```

File: wiki/PlaceholderExpansion.md [hunk 10]
```diff
@@ -58,19 +58,27 @@ public class SomeExpansion extends PlaceholderExpansion {
...
+
 ----
 ## Without a Plugin
 A PlaceholderExpansion may not need a plugin to rely on, if the placeholders it provides can return values from just the server itself or some other source (i.e. Java itself).
```
Comment (Starmism) [inline wiki/PlaceholderExpansion.md:84]:
```suggestion
An expansion does not always need a plugin to rely on. If the placeholders it provides can return values from just the server itself or some other source (i.e. Java itself), then it can work independently.
 
```

File: wiki/PlaceholderExpansion.md [hunk 11]
```diff
@@ -81,10 +89,8 @@ Common examples of such Expansions are:
 - [Server Expansion][serverexpansion]
 - [Math Expansion][mathexpansion]
 
-These kinds of Expansions don't need much more to properly work.  
-In fact is the only thing they need to have added the `onRequest(OfflinePlayer, String)` or `onPlaceholderRequest(Player, String)` methods.
-
-It is recommended to use the `onRequest(OfflinePlayer, String)` method as OfflinePlayers can also be used for normal players, but also allow to have no Player provided.
+These kinds of Expansions don't require any additional plugins to function.
```
Comment (Starmism) [inline wiki/PlaceholderExpansion.md:92]:
```suggestion
These kinds of expansions don't require any additional plugins to function.
```

File: wiki/PlaceholderExpansion.md [hunk 12]
```diff
@@ -130,7 +136,7 @@ public class SomeExpansion extends PlaceholderExpansion {
 
 ----
 ## With a Plugin (External Jar)
-If your Expansion relies on a Plugin to provide its placeholder values will you need to override a few more methods to make everything work as it should.
+If your Expansion relies on a Plugin to provide its placeholder values you will need to override a few more methods to make sure everything will work as it should.
```
Comment (Starmism) [inline wiki/PlaceholderExpansion.md:139]:
```suggestion
If your expansion relies on a plugin to provide its placeholder values, you will need to override a few more methods to make sure everything will work correctly.
```

File: wiki/PlaceholderExpansion.md [hunk 13]
```diff
@@ -130,7 +136,7 @@ public class SomeExpansion extends PlaceholderExpansion {
...
 
 Your expansion will need to override the `getRequiredPlugin()` method to return the name of the plugin your expansion depends on.  
 PlaceholderAPI does by default check, if this method either returns null, or the name defined results in a Not-null plugin being available.
```
Comment (Starmism) [inline wiki/PlaceholderExpansion.md:142]:
```suggestion
PlaceholderAPI automatically checks if this method will either return null, or if the name defined results in a non-null plugin.
```

File: wiki/PlaceholderExpansion.md [hunk 14]
```diff
@@ -58,19 +58,27 @@ public class SomeExpansion extends PlaceholderExpansion {
 Let's quickly break down the different methods you had to import.
```
Comment (Starmism) [inline wiki/PlaceholderExpansion.md:58]:
```suggestion
Let's quickly break down the different methods you have to implement.
```

**Review by Starmism — CHANGES_REQUESTED** [review]
Did a quick native-English run-through. It wouldn't let me do any more, so I'm probably going to be doing another one :)

File: wiki/PlaceholderExpansion.md [hunk 15]
```diff
@@ -1,397 +1,289 @@
...
-     * @return The name of the author as a String.
-     */
+----
+## With a Plugin (External Jar)
+If your Expansion relies on a Plugin to provide its placeholder values you will need to override a few more methods to make sure everything will work as it should.
+
+Your expansion will need to override the `getRequiredPlugin()` method to return the name of the plugin your expansion depends on.  
+PlaceholderAPI does by default check, if this method either returns null, or the name defined results in a Not-null plugin being available.
+
+Something worth noting is, that it is a bit more difficult to make a separate Jar file that depends on a plugin, as it will require the plugin to have some sort of accessible API to use in order to get the values needed.
```
Comment (Starmism) [inline wiki/PlaceholderExpansion.md:144]:
```suggestion
It is worth noting that it is a bit more difficult to make a separate jar file that depends on a plugin, as it will require the plugin to have some sort of accessible API in order to get the required values.
```

File: wiki/PlaceholderExpansion.md [hunk 16]
```diff
@@ -1,397 +1,289 @@
...
+```java
+SomePlugin plugin; // This would be the plugin your expansion depends on
+
+@Override
+public boolean canregister(){
+    // This sets plugin to the SomePlugin instance you get through the PluginManager
+    return (plugin = (SomePlugin) Bukkit.getPluginManager().getPlugin(getRequiredPlugin())) != null;
+}
+```
+With this code-snippet can you get a direct instance of the Plugin and access things such as config values.
```
Comment (Starmism) [inline wiki/PlaceholderExpansion.md:156]:
```suggestion
Using this code-snippet, you can get a direct instance of the plugin and access things such as config values.
```

File: wiki/PlaceholderExpansion.md [hunk 17]
```diff
@@ -1,397 +1,289 @@
...
+SomePlugin plugin; // This would be the plugin your expansion depends on
+
+@Override
+public boolean canregister(){
+    // This sets plugin to the SomePlugin instance you get through the PluginManager
+    return (plugin = (SomePlugin) Bukkit.getPluginManager().getPlugin(getRequiredPlugin())) != null;
+}
+```
+With this code-snippet can you get a direct instance of the Plugin and access things such as config values.  
+With that said is it recommended to instead use an API if available, as this kind of plugin access is not really a good approach.
```
Comment (Starmism) [inline wiki/PlaceholderExpansion.md:157]:
```suggestion
With that said, it is recommended instead to use an API if one is available, as this kind of plugin access is a relatively poor approach.
```

File: wiki/PlaceholderExpansion.md [hunk 18]
```diff
@@ -1,397 +1,289 @@
...
+            return plugin.getConfig().getString("placeholders.placeholder2", "default2");
+        
+        return null; // Placeholder is unknown by the Expansion
     }
 }
 ```
 
-As you can see is this method pretty similar to the one without any external plugins, since we can get an instance of our plugin much easier and also have a 100% guarantee that the plugin is installed and running.
+### Register the Expansion
+To now register the expansion will you need to call the `register()` method yourself.
```
Comment (Starmism) [inline wiki/PlaceholderExpansion.md:284]:
```suggestion
To register the expansion, you will need to call the `register()` method yourself.
```

File: wiki/PlaceholderExpansion.md [hunk 19]
```diff
@@ -1,397 +1,289 @@
...
+        
+        return null; // Placeholder is unknown by the Expansion
     }
 }
 ```
 
-As you can see is this method pretty similar to the one without any external plugins, since we can get an instance of our plugin much easier and also have a 100% guarantee that the plugin is installed and running.
+### Register the Expansion
+To now register the expansion will you need to call the `register()` method yourself.  
+This should be done in your plugin's `onEnable()` void after you made sure that PlaceholderAPI is installed and enabled.
```
Comment (Starmism) [inline wiki/PlaceholderExpansion.md:285]:
```suggestion
This should be done in your plugin's `onEnable()` method after you make sure that PlaceholderAPI is installed and enabled.
```

File: wiki/PlaceholderExpansion.md [hunk 20]
```diff
@@ -409,3 +301,73 @@ public class SomePlugin extends JavaPlugin{
     }
 }
 ```
+
+----
+## Relational Placeholders
+Relational Placeholders are a bit more specific compared to the previous examples.  
+While they do use the same [common parts](#common-parts) like the other examples do they have a different method to return placeholders.
```
Comment (Starmism) [inline wiki/PlaceholderExpansion.md:308]:
```suggestion
While they do use the same [common parts](#common-parts) that the other examples do, they have a different method to return placeholders.
```

File: wiki/PlaceholderExpansion.md [hunk 21]
```diff
@@ -409,3 +301,73 @@ public class SomePlugin extends JavaPlugin{
...
+
+In order to use the relational placeholders feature will you need to implement the [`Relational`][relational] interface, which in return adds the `onPlaceholderRequest(Player, Player, String)` method to use.
```
Comment (Starmism) [inline wiki/PlaceholderExpansion.md:310]:
```suggestion
In order to use the relational placeholders feature, you will need to implement the [`Relational`][relational] interface, which in return adds the `onPlaceholderRequest(Player, Player, String)` method to use.
```

File: wiki/PlaceholderExpansion.md [hunk 22]
```diff
@@ -409,3 +301,73 @@ public class SomePlugin extends JavaPlugin{
...
+
+#### Full Example
+Please see the [Common parts](#common-parts) section for info on the other methods.
+
+In this Example do we use the [Internal class setup](#with-a-plugin-internal-jar) and `SomePlugin` has a `areFriends(Player, Player)` method that returns true or false based on if the Players are friends.
```
Comment (Starmism) [inline wiki/PlaceholderExpansion.md:315]:
```suggestion
In this example, we use the [Internal class setup](#with-a-plugin-internal-jar) and `SomePlugin` has an `areFriends(Player, Player)` method that returns true or false based on if the given players are friends.
```

File: wiki/PlaceholderExpansion.md [hunk 23]
```diff
@@ -409,3 +301,73 @@ public class SomePlugin extends JavaPlugin{
...
+                return ChatColor.GREEN + one.getName() + " and " + two.getName() + " are not friends!";
+        }
+        
+        return null; // Placeholder is unknown by the Expansion
+    }
+}
+```
+
+### Notes about Relational Placeholders
+Relational Placeholders will always start with `%rel_` to properly identify them as such.
```
Comment (Starmism) [inline wiki/PlaceholderExpansion.md:372]:
```suggestion
Relational placeholders will always start with `%rel_` to properly identify them.
```

File: wiki/PlaceholderExpansion.md [hunk 24]
```diff
@@ -409,3 +301,73 @@ public class SomePlugin extends JavaPlugin{
...
+        }
+        
+        return null; // Placeholder is unknown by the Expansion
+    }
+}
+```
+
+### Notes about Relational Placeholders
+Relational Placeholders will always start with `%rel_` to properly identify them as such.  
+So for the above example to work will the full placeholder need to look like `%rel_example_friend%`.
```
Comment (Starmism) [inline wiki/PlaceholderExpansion.md:373]:
```suggestion
So in the above example, the full placeholder will look like `%rel_example_friend%`.
```

File: wiki/PlaceholderExpansion.md [hunk 25]
```diff
@@ -1,397 +1,289 @@
...
-
-    /**
-     * Since we register the expansion inside our own plugin, we
-     * can simply use this method here to get an instance of our
-     * plugin.
-     *
-     * @param plugin
-     *        The instance of our plugin.
-     */
+    SomePlugin plugin;
```
Comment (Starmism) [inline wiki/PlaceholderExpansion.md:244]:
```suggestion
private SomePlugin plugin;
```

File: wiki/PlaceholderExpansion.md [hunk 26]
```diff
@@ -1,397 +1,289 @@
...
-        // %someplugin_placeholder2%
-        if(identifier.equals("placeholder2")){
-            return plugin.getConfig().getString("placeholder2", "value doesnt exist");
-        }
- 
-        // We return null if an invalid placeholder (f.e. %someplugin_placeholder3%) 
-        // was provided
-        return null;
+Finally, can you also use Dependency Injection for a more easy way to access a plugin's methods.  
+Here is a small code example of how this Dependency Injection may look like:
```
Comment (Starmism) [inline wiki/PlaceholderExpansion.md:219]:
```suggestion
Here is a small code example of how dependency injection may look:
```

File: wiki/PlaceholderExpansion.md [hunk 27]
```diff
@@ -1,397 +1,289 @@
...
 
-        // %someplugin_placeholder2%
-        if(identifier.equals("placeholder2")){
-            return plugin.getConfig().getString("placeholder2", "value doesnt exist");
-        }
- 
-        // We return null if an invalid placeholder (f.e. %someplugin_placeholder3%) 
-        // was provided
-        return null;
+Finally, can you also use Dependency Injection for a more easy way to access a plugin's methods.
```
Comment (Starmism) [inline wiki/PlaceholderExpansion.md:218]:
```suggestion
Finally, you can also use dependency injection as an easier way to access a plugin's methods.
```

File: wiki/PlaceholderExpansion.md [hunk 28]
```diff
@@ -1,397 +1,289 @@
...
+----
+## With a Plugin (Internal Jar)
+The way Expansions are handled when they are part of the plugin itself is fairly similar to when you [make a Jar without a plugin dependency](#without-a-plugin)
 
-        // %someplugin_placeholder1%
-        if(identifier.equals("placeholder1")){
-            return plugin.getConfig().getString("placeholder1", "value doesnt exist");
-        }
+In fact, you don't even have to override the `getRequiredPlugin()` and `canRegister()` methods as it is always guaranteed that the plugin is available.  
+Something worth noting is, that you need to override the `persist()` method and make it return true. This ensures that the Expansion won't be unregistered by PlaceholderAPI whenever it is reloaded.
```
Comment (Starmism) [inline wiki/PlaceholderExpansion.md:216]:
```suggestion
Something worth noting, however, is that you need to override the `persist()` method and make it return true. This ensures that the expansion won't be unregistered by PlaceholderAPI whenever it is reloaded.
```

File: wiki/PlaceholderExpansion.md [hunk 29]
```diff
@@ -1,397 +1,289 @@
...
+    }
+}
+```
 
-        if(p == null){
-            return "";
-        }
+----
+## With a Plugin (Internal Jar)
+The way Expansions are handled when they are part of the plugin itself is fairly similar to when you [make a Jar without a plugin dependency](#without-a-plugin)
```
Comment (Starmism) [inline wiki/PlaceholderExpansion.md:213]:
```suggestion
The way expansions are handled when they are part of the plugin itself is fairly similar to when you [make an expansion without a plugin dependency](#without-a-plugin).
```

File: wiki/PlaceholderExpansion.md [hunk 30]
```diff
@@ -1,397 +1,289 @@
...
     @Override
-    public String onPlaceholderRequest(Player player, String identifier){
+    public String onRequest(OfflinePlayer player, String params){
+        if(params.equalsIgnoreCase("placeholder1")){
+            return plugin.getConfig().getString("placeholders.placeholder1", "default1");
+    
+        if(params.equalsIgnoreCase("placeholder2")){
+            return plugin.getConfig().getString("placeholders.placeholder2", "default2");
+        
+        return null; // Placeholder is unknown by the Expansion
```
Comment (Starmism) [inline wiki/PlaceholderExpansion.md:206]:
```suggestion
        return null; // Placeholder is unknown by the expansion
```

File: wiki/PlaceholderExpansion.md [hunk 31]
```diff
@@ -1,397 +1,289 @@
...
+- #### getVersion
+  This is a String, which means it doesn't have to be a number in itself. The String is used to determine if a new update is available or not, when the expansion is shared on the eCloud.  
+  For Expansions that are part of another plugin does this does not really matter.
 
-        // %example_placeholder2%
-        if(identifier.equals("placeholder2")){
-            return "placeholder2 works";
-        }
+Those are all the neccessary parts for your PlaceholderExpansion.  
+Any other methods that are part of the [`PlaceholderExpansion`][placeholderexpansion] class are optional and will usually not be used, or default to a specific value. Please read the Javadoc comments of those methods for more information.
```
Comment (Starmism) [inline wiki/PlaceholderExpansion.md:72]:
```suggestion
Any other methods that are part of the [`PlaceholderExpansion`][placeholderexpansion] class are optional and will usually not be used, or will default to a specific value. Please read the Javadoc comments of those methods for more information.
```

File: wiki/PlaceholderExpansion.md [hunk 32]
```diff
@@ -1,397 +1,289 @@
...
-- [Player expansion](/PlaceholderAPI/Player-Expansion)
-- [Math expansion](https://github.com/Andre601/Math-Expansion)
-- [Statistics expansion](/PlaceholderAPI/Statistics-Expansion)
+## Getting started
+To get started, first choose what type of [`PlaceholderExpansion`][placeholderexpansion] you want to create. There are various ways you can create a [`PlaceholderExpansion`][placeholderexpansion] from which this page will cover the most common ones.
 
-Since it would be weird (and also make no real sense) to have this inside your plugin, we assume you make a separate jar-file as an expansion.
+### Common Parts
+All shown examples will share the same, common parts that belong to the [`PlaceholderExpansion`][placeholderexpansion] class.  
+To not repeat the same basic info for each method throughout this page will we cover the most basic/neccessary ones here.
```
Comment (Starmism) [inline wiki/PlaceholderExpansion.md:31]:
```suggestion
In order to not repeat the same basic info for each method throughout this page, and to greatly reduce its overall length, we will cover the most basic/necessary ones here.
``

File: wiki/PlaceholderExpansion.md [hunk 33]
```diff
@@ -1,397 +1,289 @@
...
-Examples of such expansions are:
-- [Player expansion](/PlaceholderAPI/Player-Expansion)
-- [Math expansion](https://github.com/Andre601/Math-Expansion)
-- [Statistics expansion](/PlaceholderAPI/Statistics-Expansion)
+## Getting started
+To get started, first choose what type of [`PlaceholderExpansion`][placeholderexpansion] you want to create. There are various ways you can create a [`PlaceholderExpansion`][placeholderexpansion] from which this page will cover the most common ones.
 
-Since it would be weird (and also make no real sense) to have this inside your plugin, we assume you make a separate jar-file as an expansion.
+### Common Parts
+All shown examples will share the same, common parts that belong to the [`PlaceholderExpansion`][placeholderexpansion] class.
```
Comment (Starmism) [inline wiki/PlaceholderExpansion.md:30]:
```suggestion
All shown examples will share the same common parts that belong to the [`PlaceholderExpansion`][placeholderexpansion] class.
```

File: wiki/PlaceholderExpansion.md [hunk 34]
```diff
@@ -1,397 +1,289 @@
...
+  - [Notes about Relational Placeholders](#notes-about-relational-placeholders)
 
-### Without an external plugin
-This part here covers how you create an expansion that doesn't require any external/additional plugins to function.  
-Examples of such expansions are:
-- [Player expansion](/PlaceholderAPI/Player-Expansion)
-- [Math expansion](https://github.com/Andre601/Math-Expansion)
-- [Statistics expansion](/PlaceholderAPI/Statistics-Expansion)
+## Getting started
+To get started, first choose what type of [`PlaceholderExpansion`][placeholderexpansion] you want to create. There are various ways you can create a [`PlaceholderExpansion`][placeholderexpansion] from which this page will cover the most common ones.
```
Comment (Starmism) [inline wiki/PlaceholderExpansion.md:27]:
```suggestion
For starters, you need to decide what type of [`PlaceholderExpansion`][placeholderexpansion] you want to create. There are various ways to create an expansion. This page will cover the most common ones.
```

File: wiki/PlaceholderExpansion.md [hunk 35]
```diff
@@ -1,397 +1,289 @@
...
+## Table of Contents
 
-* [Without external plugin](#without-external-plugin)
-* [With external plugin](#with-external-plugin)
-  * [Separate jar](#separate-jar)
-  * [Internal class](#internal-class)
+- [Getting started](#getting-started)
+  - [Common Parts](#common-parts)
+- [Without a Plugin](#without-a-plugin)
+- [With a Plugin (External Jar)](#with-a-plugin-external-jar)
```
Comment (Starmism) [inline wiki/PlaceholderExpansion.md:20]:
```suggestion

```

File: wiki/PlaceholderExpansion.md [hunk 36]
```diff
@@ -1,397 +1,289 @@
...
+
+[relational]: https://github.com/PlaceholderAPI/PlaceholderAPI/blob/master/src/main/java/me/clip/placeholderapi/expansion/Relational.java
 
-PlaceholderAPI is using Expansions for its placeholders, with PlaceholderAPI providing the core. Users can download Expansions from the cloud server using commands in-game or by going [here](https://api.extendedclip.com/home/) if they want to use the placeholder.
+## Overview
+This page will cover how you can create your own [`PlaceholderExpansion`][placeholderexpansion] which you can either [[Upload to the eCloud|Expansion cloud]] or integrate into your own plugin.
 
-## Note
-You can either make a separate jar file, to upload it to the expansion-cloud (recommended) or have it as a local class inside your plugin.
+Something to note here is, that PlaceholderAPI relies on Expansions to be installed. PlaceholderAPI is acting as the core while the Expansions are what allows other plugins to use placeholders in various messages.
```
Comment (Starmism) [inline wiki/PlaceholderExpansion.md:12]:
```suggestion
It's worth noting that PlaceholderAPI relies on expansions being installed. PlaceholderAPI only acts as the core replacing utility while the expansions allow other plugins to use any installed placeholder in their own messages.
```

**Review by Starmism — CHANGES_REQUESTED** [review]
Here's the rest! I'll do a final review after these are merged just to be sure. 👍

File: wiki/PlaceholderExpansion.md (hunk 35 above)
Comment (Andre601) [reply wiki/PlaceholderExpansion.md:20]: Why should this be removed?

File: wiki/PlaceholderExpansion.md (hunk 33 above)
Comment (Andre601) [reply wiki/PlaceholderExpansion.md:30]: You removed the two spaces at the end for the line break...

File: wiki/PlaceholderExpansion.md (hunk 3 above)
Comment (Andre601) [reply wiki/PlaceholderExpansion.md:64]: ** won't really work here since it's in a inline code. Also, I doubt people are THAT dumb to not recognize the part literally called "identifier"

File: wiki/PlaceholderExpansion.md (hunk 35 above)
Comment (Starmism) [reply wiki/PlaceholderExpansion.md:20]: Is it not a duplicate?

File: wiki/PlaceholderExpansion.md (hunk 3 above)
Comment (Starmism) [reply wiki/PlaceholderExpansion.md:64]: Ah okay, can keep the identifier normal then, but I would suggest implementing the rest of the changes!

File: wiki/PlaceholderExpansion.md (hunk 35 above)
Comment (Andre601) [reply wiki/PlaceholderExpansion.md:20]: Can you not read?

Comment (Starmism) [reply wiki/PlaceholderExpansion.md:20]: Oh no those look extremely similar, and I wouldn't necessarily call it an `Internal Jar` which is probably why I never looked at the parentheses lol

**Comment by Andre601** [comment]
Since I applied all suggested changes and didn't hear any other feedback so far would I call this PR done and merge it now to update the wiki.
Further changes can always be done of course.
